### DISCUSSION TITLE
What happens if a file with a recognized face is removed?

### DISCUSSION THREAD

**Discussion by Rolf-Smit** [photoprism/photoprism#3198] [Q&A]
I moved some files around in my originals folder (as I also organize my photo's in logical folders), however after a full re-index some thumbnails seem to be missing from the people section. I looked a bit deeper into the database and the thumbnails that are used seem to be pointing to missing files. Running the thumbs command did not resolve these missing thumbnails (neither did clean or purge, faces audit or anything similar).

**Can it be the case that when files get removed (or maybe moved for that matter), the recognized faces are not getting removed as well?**

_I also noticed that the face counts where totally off (reporting 1 face) but up on clicking on a subject I could see at least 5 photo's, but upon clicking on those photo's there where not faces shown (but the tab did show that there should be 2 faces)._

**Comment by lastzero** [comment]
Are you 100% sure that these files have not been modified in any way? This usually happens when a hash changes and the files are renamed at the same time.

If these files are PNGs or there is also a PNG version, there is also a possibility that this has to do with the newly implemented support for PNG files as primary images. This eliminates the need to create a JPEG from them first, which saves disk space and can speed up indexing.

Running the cleanup command while files are marked as missing or replacing all thumbnails and JPEGs with updated versions can also cause this.

**Comment by Rolf-Smit** [comment]
I opened a discussion since I cannot be 100% sure of the integrity of the files prior to re-indexing as I found a failing memory module had caused quite some damage to created thumbnail files (and possibly the index, but luckily my originals are elsewhere and have good back-ups).

I'm trying to figure out how PhotoPrism works on the inside to see if the index is salvageable. 

As far as I understand now: during indexing the hashes may have been calculated wrong due to the failing memory module and therefore re-indexed files (and thus also moved files) may also have changed hash, potentially causing this issue?

At this point I'm starting to think that the faces index is not salvageable on it's own, there seems to be too many weird errors. The index is self seems to be fine after a full re-index and thumbnail generation.

**Reply by lastzero** [reply]
I'm not sure if the hash calculation is affected by this particular issue, but it's possible. We did make sure that face markers are not lost even if a file is updated (and the hash changes), but there may be cases where this doesn't work for reasons we didn't think of when implementing it.
